echo/echo.py

```diff
@@ -1,131 +1,161 @@
+                self.sio.emit(
+                    "text",
+                    {"receiver_id": data["user"]["id"], "message": message},
+                    callback=self.message_callback
+                )
+
+        @self.sio.event
+        def image_message(data):
+            sender_id = data["user"]["id"]
+            if self.user is not None and self.user == sender_id:
+                return
+
+            LOG.debug(f"I got an image, let's send it back!: {data}")
+
+            if not data["private"]:
+                self.sio.emit(
+                    "image",
+                    {
+                        "room": data["room"],
+                        "url": data["url"],
+                        "width": data["width"],
+                        "height": data["height"]
+                    },
+                    callback=self.message_callback
+                )
+            else:
+                LOG.debug("It was actually a private message o.O")
+                self.sio.emit(
+                    "image",
+                    {
+                        "receiver_id": data["user"]["id"],
+                        "url": data["url"],
+                        "width": data["width"],
+                        "height": data["height"]
+                    },
+                    callback=self.message_callback
+                )
+
+
+if __name__ == "__main__":
+    # set up logging configuration
+    logging.basicConfig(level=logging.DEBUG, format="%(levelname)s:%(message)s")
+
+    # create commandline parser
+    parser = argparse.ArgumentParser(description="Run Echo Bot.")
+
+    # collect environment variables as defaults
+    if "SLURK_TOKEN" in os.environ:
+        token = {"default": os.environ["SLURK_TOKEN"]}
     else:
-        task_id = {'default': None}
-
-    parser.add_argument('-t', '--token',
-                        help='token for logging in as bot (see SERVURL/token)',
-                        **token)
-    parser.add_argument('-c', '--chat_host',
-                        help='full URL (protocol, hostname; ending with /) of chat server',
-                        **chat_host)
-    parser.add_argument('-p', '--chat_port',
-                        type=int,
-                        help='port of chat server',
-                        **chat_port)
-    parser.add_argument('--task_id',
-                        type=int,
-                        help='Task to join',
-                        **task_id)
+        token = {"required": True}
+    if "SLURK_USER" in os.environ:
+        user = {"default": os.environ["SLURK_USER"]}
+    else:
+        user = {"required": True}
+    host = {"default": os.environ.get("SLURK_HOST", "http://localhost")}
+    port = {"default": os.environ.get("SLURK_PORT")}
+    task_id = {"default": os.environ.get("ECHO_TASK_ID")}
+
+    # register commandline arguments
+    parser.add_argument(
+        "-t", "--token", help="token for logging in as bot (see SERVURL/token)", **token
```

Comment:
This was a long time ago 🤣 
```suggestion
        "-t", "--token", help="token for logging in as bot", **token
```